packages/sdk/typescript/human-protocol-sdk/src/staking.ts

```diff
@@ -0,0 +1,362 @@
+      throw ErrorInvalidEscrowAddressProvided;
+    }
+
+    try {
+      const rewardPoolContract: RewardPool = RewardPool__factory.connect(
+        await this.stakingContract.rewardPool(),
+        this.signerOrProvider
+      );
+
+      await rewardPoolContract.distributeReward(escrowAddress);
+      return;
+    } catch (e) {
+      throw ErrorStakingFailedToDistributeRewards;
+    }
+  }
+
+  /**
+   * **Returns the staking information about an staker address.*
+   *
+   * @param {string} staker - Address of the staker
+   * @returns {Promise<IStakerInfo>} - Return staking information of the specified address
+   * @throws {Error} - An error object if an error occurred, result otherwise
+   */
+  public async getStaker(staker: string): Promise<IStaker> {
+    if (!ethers.utils.isAddress(staker)) {
+      throw ErrorInvalidStakerAddressProvided;
+    }
+
+    try {
+      const result = await this.stakingContract.getStaker(staker);
+      return result;
+    } catch (e) {
+      console.error(e);
+      throw ErrorStakingGetStaker;
+    }
+  }
+
+  /**
+   * **Returns the staking information about all stakers of the protocol.*
+   *
+   * @returns {Promise<IStakerInfo>} - Return an array with all stakers information
+   * @throws {Error} - An error object if an error occurred, results otherwise
+   */
+  public async getAllStakers(): Promise<IStaker[]> {
+    try {
+      const result = await this.stakingContract.getListOfStakers();
+
+      if (result[1].length === 0) {
+        throw ErrorStakingStakersNotFound;
+      }
+
+      return result[1];
+    } catch (e) {
+      console.error(e);
+      throw ErrorStakingStakersNotFound;
+    }
+  }
+
+  /**
+   * **Returns information about the allocation of the specified escrow.*
+   *
+   * @param {string} escrowAddress - The escrow address for the received allocation data
+   * @returns {Promise<IAllocation>} - Returns allocation info if exists
+   * @throws {Error} - An error object if an error occurred, result otherwise
+   */
+  public async getAllocation(escrowAddress: string): Promise<IAllocation> {
+    if (!ethers.utils.isAddress(escrowAddress)) {
+      throw ErrorInvalidEscrowAddressProvided;
+    }
+
+    try {
+      const result = await this.stakingContract.getAllocation(escrowAddress);
+      return result;
+    } catch (e) {
+      console.error(e);
+      throw ErrorStakingGetAllocation;
+    }
+  }
+
+  /**
+   * **Returns information about the rewards for a given escrow address.*
+   *
+   * @param {string} escrowAddress - Address of the escrow
+   * @returns {Promise<IReward[]>} - Returns rewards info if exists
+   * @throws {Error} - An error object if an error occurred, results otherwise
+   */
+  public async getRewards(escrowAddress: string): Promise<IReward[]> {
+    if (!ethers.utils.isAddress(escrowAddress)) {
+      throw ErrorInvalidEscrowAddressProvided;
+    }
+
+    try {
+      const rewardPoolContract: RewardPool = RewardPool__factory.connect(
```

Comment:
this should be in the constructor and add rewardpool address to constants

Reply:
1. Reward pool address saved to the staking contract and can be changed any time, I think incorrect to add it as a constants
2. Impossible to put it to the constructor because it is async method 

Probably to simplify it, we can save it as a constants, but we have to be careful

Reply:
I dont see a reason to change the reward pool, but it is ok, lets keep it as you have it now